From 6e37df34dfeea838560795e0ac68380773141d51 Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Fri, 10 Aug 2018 10:06:46 +0800
Subject: [PATCH 2/2] driver: ethernet:mvpp2x: initialize spinlock

There is below call trace when kernel boots up with kts test template.

Workqueue: mv_pp2x mv_pp2x_get_device_stats Call trace:
[<ffffff800808bef0>] dump_backtrace+0x0/0x230
[<ffffff800808c144>] show_stack+0x24/0x30
[<ffffff8008576000>] dump_stack+0xac/0xe4
[<ffffff80080fe62c>] register_lock_class+0x3a4/0x5b8
[<ffffff800810202c>] __lock_acquire+0x84/0x19d0
[<ffffff8008104108>] lock_acquire+0xe8/0x2d0
[<ffffff8008a6c438>] _raw_spin_lock_irqsave+0x58/0x98
[<ffffff8008627210>] mv_gop110_mib_counters_stat_update+0x80/0x460
[<ffffff8008617640>] mv_pp2x_get_device_stats+0xf0/0x108
[<ffffff80080c7410>] process_one_work+0x298/0x7a0
[<ffffff80080c7968>] worker_thread+0x50/0x480
[<ffffff80080cf280>] kthread+0x138/0x140
[<ffffff8008085ea0>] ret_from_fork+0x10/0x18

Because there is no code to initialize field stats_spinlock of struct
mv_mac_data. So initialize the spinlock for every Ethernet port.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 87e4172..fdb3ab7 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -5330,6 +5330,8 @@ static int mv_pp2x_port_probe(struct platform_device *pdev,
 
 	mv_pp2x_port_irq_names_update(port);
 
+	spin_lock_init(&port->mac_data.stats_spinlock);
+
 	netdev_info(dev, "Using %s mac address %pM\n", mac_from, dev->dev_addr);
 
 	priv->port_list[priv->num_ports] = port;
-- 
1.7.9.5

